### DISCUSSION TITLE
react/dns why limit udp max packet size?

### DISCUSSION THREAD

**Discussion by v1talM** [reactphp/reactphp#575] [Q&A] [answered]
I found that **UdpTransportExecutor** limit **maxPacketSize=512;**, and no setter() in this class. 

why not make this param editable?😄

**Comment by WyriHaximus** [comment]
Well mainly because the maximum size a DNS package over UDP can be is 512 bytes as per RFC: https://www.rfc-editor.org/std/std75.txt#:~:text=DNS%20%5BRFC1035%5D%20specifies%20a%20message,this%20document%20is%20512%20bytes.

**Reply by v1talM** [reply]
thanks! 😄
I wonder if DNS Server return  EDNS(0) results with packet size over than 512 bytes, whether UdpTransportExecutor will works?

**Reply by WyriHaximus** [reply, marked as answer]
It shouldn't, but that's why we set up both at https://github.com/reactphp/dns/blob/3.x/src/Resolver/Factory.php#L164-L180 and then when that situation happens we try over TCP instead: https://github.com/reactphp/dns/blob/3.x/src/Query/SelectiveTransportExecutor.php#L69-L78
